runtime/queries/c-sharp/context.scm

```diff
@@ -0,0 +1,64 @@
+; Credits to nvim-treesitter-context
+
+(interface_declaration
+  body: (_) @context.params
+) @context
+
+(class_declaration
+  body: (_) @context.params
+) @context
+
+(enum_declaration
+  body: (_) @context.params
+) @context
+
+(struct_declaration
+  body: (_) @context.params
+) @context
+
+(record_declaration
+  body: (_) @context.params
+) @context
+
+(namespace_declaration
+  body: (_) @context.params
+) @context
+
+(constructor_declaration
+  body: (_) @context.params
+) @context
+
+(destructor_declaration
+  body: (_) @context.params
+) @context
+
+(method_declaration
+  body: (_) @context.params
+) @context
+
+(switch_statement
+  body: (_) @context.params
+) @context
+
+(for_statement
+  body: (_) @context.params
+) @context
+
+(if_statement
+  consequence: (_) @context.params
+) @context
+
+([
+  (do_statement)
+  (while_statement)
+] @context)
+
+(try_statement
+  body: (_) @context.params
+) @context
+
+(catch_clause
+  body: (_) @context.params
+) @context
+
+(finally_clause) @context
```

Comment:
```suggestion
(interface_declaration) @context

(class_declaration) @context

(enum_declaration) @context

(struct_declaration) @context

(record_declaration) @context

(namespace_declaration) @context

(constructor_declaration
    (parameter_list) @context.params
  ) @context

(destructor_declaration
    (parameter_list) @context.params
  ) @context

(method_declaration
    (parameter_list) @context.params
  ) @context

(switch_statement) @context

(for_statement) @context

(if_statement) @context

([
  (do_statement)
  (while_statement)
] @context)

(try_statement) @context

(catch_clause) @context

(finally_clause) @context

```

the `@context.params` aren't used for most of the definitions, only for methods it may be important.

But also to that: i am contemplating whether or not it is actually good to have `@context.params` considering that with the weird attributes you can have in C# we have issues like this: 
![image](https://github.com/SoraTenshi/helix/assets/13474089/3d9571ef-22bb-4bf8-a6af-e303c3c955e1)

Reply:
hm i see. I'll improve my tree-sitter knowledge to check better way to write rules